tsdb/head.go

```diff
@@ -1118,12 +1120,38 @@ func (a *headAppender) Add(lset labels.Labels, t int64, v float64) (uint64, erro
 		return 0, err
 	}
 
+	meta := false
+	s.Lock()
+	if s.typ != m.Type {
```

Comment:
this block will result in whatever the last seen metadata for a series (based on labelset hash) was being used as the metadata for an entire block, is that right?